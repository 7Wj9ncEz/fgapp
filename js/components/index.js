import App from './app';
import MainMenu from './main-menu';
import DrawerMenu from './drawer-menu';
import AppRoutes from './app-routes';
import FgaNews from './fga-news';
import Events from './events';
import Article from './article';
import FgaProfessors from './fga-professors';
import Professor from './professor';
import NewsListItem from './news-list-item';
import ProfessorsListItem from './professors-list-item';

export {
  App,
  Article,
  AppRoutes,
  MainMenu,
  DrawerMenu,
  FgaNews,
  Events,
  FgaProfessors,
  Professor,
  NewsListItem,
  ProfessorsListItem
};
