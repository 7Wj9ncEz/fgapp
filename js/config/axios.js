import axios from 'axios';

export const BASE_URL = 'https://fga.unb.br';
export const API_URL = `${BASE_URL}/api/v1`;
export const TIMEOUT = 10000;

const axiosInstance = axios.create({
  baseURL: API_URL,
  timeout: TIMEOUT
});

export default axiosInstance;
