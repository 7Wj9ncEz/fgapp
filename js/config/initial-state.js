import {
  ALL,
  AEROESPACIAL,
  AUTOMOTIVA,
  ELETRONICA,
  ENERGIA,
  SOFTWARE
} from './professor-communities';

export const COMMON_DEFAULTS = {
  page: 1,
  lastPage: Infinity,
  data: []
}

export default {
  articles: {
    page: COMMON_DEFAULTS.page,
    lastPage: COMMON_DEFAULTS.lastPage,
    data: COMMON_DEFAULTS.data,
    error: false
  },

  professors: {
    currentCourse: ALL,
    error: false,

    [ALL]: {
      data: COMMON_DEFAULTS.data,
      page: COMMON_DEFAULTS.page,
      lastPage: COMMON_DEFAULTS.lastPage
    },

    [AEROESPACIAL]: {
      data: COMMON_DEFAULTS.data,
      page: COMMON_DEFAULTS.page,
      lastPage: COMMON_DEFAULTS.lastPage
    },

    [AUTOMOTIVA]: {
      data: COMMON_DEFAULTS.data,
      page: COMMON_DEFAULTS.page,
      lastPage: COMMON_DEFAULTS.lastPage
    },

    [ELETRONICA]: {
      data: COMMON_DEFAULTS.data,
      page: COMMON_DEFAULTS.page,
      lastPage: COMMON_DEFAULTS.lastPage
    },

    [ENERGIA]: {
      data: COMMON_DEFAULTS.data,
      page: COMMON_DEFAULTS.page,
      lastPage: COMMON_DEFAULTS.lastPage
    },

    [SOFTWARE]: {
      data: COMMON_DEFAULTS.data,
      page: COMMON_DEFAULTS.page,
      lastPage: COMMON_DEFAULTS.lastPage
    }
  },

  pages: {
    actualPage: 'FgaNews'
  }
};