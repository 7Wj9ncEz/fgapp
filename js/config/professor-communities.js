export const ALL = "35"
export const AEROESPACIAL = "38"
export const AUTOMOTIVA = "41"
export const ENERGIA = "39"
export const ELETRONICA = "40"
export const SOFTWARE = "37"
export const communitiesId = {ALL, AEROESPACIAL, AUTOMOTIVA, ENERGIA, ELETRONICA, SOFTWARE}

export default communitiesId
