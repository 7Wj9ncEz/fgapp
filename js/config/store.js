import { createStore, combineReducers } from 'redux';

import articlesReducer from '../reducers/articles-reducer';
import professorsReducer from '../reducers/professors-reducer';

export const initialState = {
  articles: {
    page: 1,
    lastPage: Infinity,
    data: [],
    error: false
  },
  professors: []
};


const combinedReducers = combineReducers({
  articles: articlesReducer,
  professors: professorsReducer
});

const store = createStore(combinedReducers, initialState);

export default store;
