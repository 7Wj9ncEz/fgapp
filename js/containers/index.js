import FgaNewsContainer from './fga-news-container';

export {
  FgaNewsContainer
}