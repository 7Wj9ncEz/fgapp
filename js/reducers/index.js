import articlesReducer from './articles-reducer';
import professorsReducer from './professors-reducer';

export {
  articlesReducer,
  professorsReducer
}