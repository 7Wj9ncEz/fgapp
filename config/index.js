import axios from 'axios';

export const axiosInstance = axios.create({
  baseURL: 'https://fga.unb.br/api/v1'
});
