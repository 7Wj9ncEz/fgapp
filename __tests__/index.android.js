import 'react-native';
import React from 'react';
import FGApp from '../index.android.js';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import { store } from './../js/config';
import { Provider } from 'react-redux';

it('renders correctly', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <FGApp/>
    </Provider>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
