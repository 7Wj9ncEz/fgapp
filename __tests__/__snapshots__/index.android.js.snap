exports[`test renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }>
  <View
    style={
      Object {
        "flex": 9,
      }
    }>
    <View
      onLayout={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "flex": 1,
          },
          undefined,
        ]
      }>
      <View
        style={
          Object {
            "flex": 1,
          }
        }>
        <View
          onMoveShouldSetResponder={[Function]}
          onMoveShouldSetResponderCapture={[Function]}
          onResponderEnd={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderReject={[Function]}
          onResponderRelease={[Function]}
          onResponderStart={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          onStartShouldSetResponderCapture={[Function]}
          pointerEvents="auto"
          style={
            Object {
              "backgroundColor": "white",
              "bottom": 0,
              "left": 0,
              "opacity": 1,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "transform": Array [
                Object {
                  "translateX": 0,
                },
                Object {
                  "translateY": 0,
                },
              ],
            }
          }>
          <View
            style={
              Array [
                Object {
                  "flex": 1,
                },
                undefined,
              ]
            }>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "yellow",
                  "flex": 1,
                  "justifyContent": "center",
                }
              }>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail">
                News
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "flex": 1,
      }
    }>
    <View
      style={
        Object {
          "backgroundColor": "#efefef",
          "borderTopWidth": 1,
          "flex": 1,
          "flexDirection": "row",
        }
      }>
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
          }
        }>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hitSlop={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID={undefined}>
          <Text
            accessible={true}
            allowFontScaling={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#333",
                  "fontSize": 60,
                },
                undefined,
                Object {
                  "fontFamily": "FontAwesome",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
              ]
            }>
            
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
          }
        }>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hitSlop={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID={undefined}>
          <Text
            accessible={true}
            allowFontScaling={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#333",
                  "fontSize": 60,
                },
                undefined,
                Object {
                  "fontFamily": "FontAwesome",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
              ]
            }>
            
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
          }
        }>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hitSlop={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID={undefined}>
          <Text
            accessible={true}
            allowFontScaling={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#333",
                  "fontSize": 60,
                },
                undefined,
                Object {
                  "fontFamily": "FontAwesome",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
              ]
            }>
            
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
